src/utils/transaction/serializeTransaction.ts

```diff
@@ -137,6 +149,70 @@ export function serializeTransaction<
   ) as SerializedTransactionReturnType<transaction>
 }
 
+type SerializeTransactionRIP7560ErrorType =
+  | AssertTransactionRIP7560ErrorType
+  | ConcatHexErrorType
+  | InvalidLegacyVErrorType
+  | ToHexErrorType
+  | ToRlpErrorType
+  | ErrorType
+
+function serializeTransactionRIP7560(
+  transaction: TransactionSerializableRIP7560,
+  authorizationData?: Hex | undefined,
+): TransactionSerializedRIP7560 {
+  const {
+    chainId,
+    gas,
+    nonce,
+    accessList,
+    maxFeePerGas,
+    maxPriorityFeePerGas,
+    nonceKey,
+    sender,
+    executionData,
+    builderFee,
+    verificationGasLimit,
+    deployer,
+    deployerData,
+    paymaster,
+    paymasterData,
+    paymasterVerificationGasLimit,
+    paymasterPostOpGasLimit,
+  } = transaction
+
+  assertTransactionRIP7560(transaction)
+
+  const serializedAccessList = serializeAccessList(accessList)
+
+  const serializedTransaction = [
+    toHex(chainId),
+    nonce ? toHex(nonce) : '0x',
+    nonceKey ? toHex(nonceKey) : '0x',
+    sender,
+    deployer ?? '0x',
+    deployerData ?? '0x',
+    paymaster ?? '0x',
+    paymasterData ?? '0x',
+    executionData,
+    builderFee ? toHex(builderFee) : '0x',
+    maxPriorityFeePerGas ? toHex(maxPriorityFeePerGas) : '0x',
+    maxFeePerGas ? toHex(maxFeePerGas) : '0x',
+    toHex(verificationGasLimit),
+    paymasterVerificationGasLimit ? toHex(paymasterVerificationGasLimit) : '0x',
+    paymasterPostOpGasLimit ? toHex(paymasterPostOpGasLimit) : '0x',
+    gas ? toHex(gas) : '0x',
+    serializedAccessList,
+  ]
+
+  if (authorizationData) serializedTransaction.push(authorizationData)
```

Comment:
I think it can be modified like this
```suggestion
    serializedAccessList,
    ...(authorizationData ? [authorizationData] : [])
  ]
```

Reply:
But this will change the result of RLP encoding due to the empty array?